adapter/repository/in-memory-simple/src/test/java/org/mobilitydata/gtfsvalidator/db/InMemoryGtfsDataRepositoryTest.java

```diff
@@ -447,4 +443,49 @@ void addFareRuleAndGetFareRuleShouldReturnSameEntity() {
         assertEquals(mockFareRule01, underTest.getFareRule("fare id1", null, null,
                 null, null));
     }
+
+
+    @Test
+    void callToAddShapeShouldSameReturnEntity() {
+        final ShapePoint mockShapePoint = mock(ShapePoint.class);
+        when(mockShapePoint.getShapeId()).thenReturn("test id");
+        final InMemoryGtfsDataRepository underTest = new InMemoryGtfsDataRepository();
+
+        assertEquals(mockShapePoint, underTest.addShapePoint(mockShapePoint));
+    }
+
+    @Test
+    void addTwiceSameShapePointShouldReturnNull() {
+        final ShapePoint mockShapePoint = mock(ShapePoint.class);
+        when(mockShapePoint.getShapeId()).thenReturn("test id");
+        final InMemoryGtfsDataRepository underTest = new InMemoryGtfsDataRepository();
+
+        underTest.addShapePoint(mockShapePoint);
+        assertNull(underTest.addShapePoint(mockShapePoint));
+    }
+
+    @Test
+    void addNullShapePointShouldThrowException() {
+        final InMemoryGtfsDataRepository underTest = new InMemoryGtfsDataRepository();
+        final Exception exception = assertThrows(IllegalArgumentException.class,
+                () -> underTest.addShapePoint(null));
+        assertEquals("Cannot add null shape point to data repository", exception.getMessage());
+    }
+
+    @Test
+    void getShapeByIdShouldReturnRelatedListOfShapePoint() {
```

Comment:
There should be additional tests to test adding points with `shape_pt_sequence` out of order, and ensuring that you get them back in-order.